libraries/tests/mbed/serial_nc_rx/main.cpp

```diff
@@ -2,29 +2,36 @@
 #include "test_env.h"
 
 int main() {
+    Serial *pc = new Serial(USBTX, USBRX);
+
     MBED_HOSTTEST_TIMEOUT(20);
     MBED_HOSTTEST_SELECT(serial_nc_rx_auto);
     MBED_HOSTTEST_DESCRIPTION(Serial NC RX);
     MBED_HOSTTEST_START("MBED_37");
 
-    Serial *pc = new Serial(NC, USBRX);
-
     char c = pc->getc();
 
-    delete pc;
 
-    // This should be true
-    if (c == 'E') {
-      Serial *pc = new Serial(USBTX, NC);
-
-      pc->printf("RX OK - Expected\r\n");
+    // This should be true, sync the start of test
+    if (c == 'S') {
```

Comment:
Yeah, in this case you'd want to know which event is expected where, so two distinct chars are used